src/base.rs

```diff
@@ -579,6 +579,25 @@ pub enum Error {
     Protocol(ProtocolError),
 }
 
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::Connection(_) => f.write_str("xcb connection error"),
+            Error::Protocol(_) => f.write_str("xcb protocol error"),
+        }
+    }
+}
+
+impl std::error::Error for Error {
+    fn cause(&self) -> Option<&dyn std::error::Error> {
```

Comment:
according to the [docs](https://doc.rust-lang.org/std/error/trait.Error.html#method.cause), `cause` is deprecated in favor of `source`

Reply:
oh right, i'm sorry, fixed